core/trino-main/src/main/java/io/trino/testing/TransactionBuilder.java

```diff
@@ -137,9 +139,14 @@ public <T> T execute(Session session, Function<Session, T> callback)
         requireNonNull(session, "session is null");
         requireNonNull(callback, "callback is null");
 
-        session = Session.builder(session)
-                .setQueryId(QUERY_ID_GENERATOR.createNextQueryId())
-                .build();
+        Session.SessionBuilder sessionBuilder = Session.builder(session)
+                .setQueryId(QUERY_ID_GENERATOR.createNextQueryId());
+        if (session.getSystemProperties().containsKey(SESSION_START_TIME_PROPERTY)) {
+            Instant sessionStartTime = session.getSystemProperty(SESSION_START_TIME_PROPERTY, Instant.class);
```

Comment:
Does this obsolete following?
https://github.com/trinodb/trino/blob/3452189dfa4d8f6b1f5bacddf3bcee84c7aba672/core/trino-main/src/main/java/io/trino/server/testing/TestingTrinoServer.java#L879-L882